### PR TITLE
Warnings 202307

### LINKED ISSUE #988: Reduce Warnings
Tired of seeing these. Let's try to address the ones that are in our code base.  We need to continually do this...

I have submitted a patch to finifft for the ones in `test_fourier_correlation` (`warnings` bug in their code), so that will get fixed eventually.

```
../../opt/miniconda3/envs/diag_matrix/lib/python3.8/site-packages/ray/_private/parameter.py:4
../../opt/miniconda3/envs/diag_matrix/lib/python3.8/site-packages/ray/_private/parameter.py:4
../../opt/miniconda3/envs/diag_matrix/lib/python3.8/site-packages/ray/_private/parameter.py:4
  /Users/gbwright/opt/miniconda3/envs/diag_matrix/lib/python3.8/site-packages/ray/_private/parameter.py:4: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources

../../opt/miniconda3/envs/diag_matrix/lib/python3.8/site-packages/pkg_resources/__init__.py:2871
../../opt/miniconda3/envs/diag_matrix/lib/python3.8/site-packages/pkg_resources/__init__.py:2871
../../opt/miniconda3/envs/diag_matrix/lib/python3.8/site-packages/pkg_resources/__init__.py:2871
  /Users/gbwright/opt/miniconda3/envs/diag_matrix/lib/python3.8/site-packages/pkg_resources/__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('mpl_toolkits')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

../../opt/miniconda3/envs/diag_matrix/lib/python3.8/site-packages/pkg_resources/__init__.py:2871: 18 warnings
  /Users/gbwright/opt/miniconda3/envs/diag_matrix/lib/python3.8/site-packages/pkg_resources/__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('sphinxcontrib')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

tests/test_fourier_correlation.py::test_frc_plot[method=nufft-img_size=64-dtype=<class 'numpy.float64'>]
tests/test_fourier_correlation.py::test_frc_plot[method=nufft-img_size=64-dtype=<class 'numpy.float32'>]
tests/test_fourier_correlation.py::test_frc_plot[method=nufft-img_size=65-dtype=<class 'numpy.float64'>]
tests/test_fourier_correlation.py::test_frc_plot[method=nufft-img_size=65-dtype=<class 'numpy.float32'>]
tests/test_fourier_correlation.py::test_fsc_plot[method=nufft-img_size=64-dtype=<class 'numpy.float64'>]
tests/test_fourier_correlation.py::test_fsc_plot[method=nufft-img_size=64-dtype=<class 'numpy.float32'>]
tests/test_fourier_correlation.py::test_fsc_plot[method=nufft-img_size=65-dtype=<class 'numpy.float64'>]
tests/test_fourier_correlation.py::test_fsc_plot[method=nufft-img_size=65-dtype=<class 'numpy.float32'>]
  /Users/gbwright/work/ASPIRE-Python.diag_matrix/src/aspire/utils/resolution_estimation.py:365: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.
    plt.show()

tests/test_image.py: 51 warnings
  /Users/gbwright/work/ASPIRE-Python.diag_matrix/tests/test_image.py:30: DeprecationWarning: scipy.misc.face has been deprecated in SciPy v1.10.0; and will be completely removed in SciPy v1.12.0. Dataset methods have moved into the scipy.datasets module. Use scipy.datasets.face instead.
    im_np = misc.face(gray=True).astype(dtype)[np.newaxis, :size, :size]

tests/test_simulation_metadata.py::SimTestCase::test_update_properties
  /Users/gbwright/work/ASPIRE-Python.diag_matrix/src/aspire/utils/rotation.py:81: UserWarning: Gimbal lock detected. Setting third angle to zero since it is not possible to uniquely determine all angles.
    euler_angles = rotations.as_euler(self._seq_order, degrees=False).astype(

tests/test_starfileio.py: 11 warnings
  /Users/gbwright/work/ASPIRE-Python.diag_matrix/tests/test_starfileio.py:50: DeprecationWarning: scipy.misc.face has been deprecated in SciPy v1.10.0; and will be completely removed in SciPy v1.12.0. Dataset methods have moved into the scipy.datasets module. Use scipy.datasets.face instead.
    self.im = Image(misc.face(gray=True).astype("float64")[:L, :L])

tests/test_weighted_mean_estimator.py::WeightedVolumesEstimatorTestCase::testOptimize2
  /Users/gbwright/work/ASPIRE-Python.diag_matrix/src/aspire/reconstruction/kernel.py:197: ComplexWarning: Casting complex values to real discards the imaginary part
    xx[k, j] = FourierKernel(self.kermat[k, j]).circularize()

tests/test_covar3d.py::Covar3DTestCase::testClustering
tests/test_covar3d.py::Covar3DTestCase::testEigsEvaluation
  /Users/gbwright/work/ASPIRE-Python.diag_matrix/src/aspire/utils/matrix.py:404: DeprecationWarning: Keyword argument 'eigvals' is deprecated in favour of 'subset_by_index' keyword instead and will be removed in SciPy 1.12.0.
    w, v = eigh(A.astype(np.float64), eigvals=(sig_len - 1 - k + 1, sig_len - 1))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

### PR DESCRIPTION
**Pull request by garrettwrong**
Closes #988

### CODE REVIEW

File: src/aspire/utils/rotation.py [hunk 1]
```diff
@@ -72,12 +72,10 @@ def angles(self):
...
 
-        # If `about_axis` was used to generate rotations about the z-axis
-        # turn off Gimbal lock warning.
+        # Filter Gimbal lock warnings.
         with warnings.catch_warnings():
-            if self._about_z:
-                msg = "Gimbal lock detected*"
-                warnings.filterwarnings("ignore", message=msg, category=UserWarning)
+            msg = "Gimbal lock detected*"
+            warnings.filterwarnings("ignore", message=msg, category=UserWarning)
```
Comment (j-c-c) [inline src/aspire/utils/rotation.py]: Do we never want to be warned about gimbal lock? If so, I'm fine with this. If not, we could just change the updated rotations in the test.

**Review by j-c-c — COMMENTED** [review]
Looks good. Just one question.

File: src/aspire/utils/rotation.py (hunk 1 above)
Comment (garrettwrong) [reply src/aspire/utils/rotation.py]: I think they are going to keep occurring whenever we do trivial about axis based tests... All we did before was carve out the default axis to ignore.

Comment (garrettwrong) [reply src/aspire/utils/rotation.py]: I can make the warnings filter optional with default to not filter and change all our test cases to filter.

That would warn users (but might complicate tutorials).

Comment (j-c-c) [reply src/aspire/utils/rotation.py]: Let's keep it like this.

**Review by j-c-c — DISMISSED** [review]

File: src/aspire/utils/rotation.py (hunk 1 above)
Comment (garrettwrong) [reply src/aspire/utils/rotation.py]: lol, too late I'm testing the changes. More code to review :D

Comment (j-c-c) [reply src/aspire/utils/rotation.py]: Lol, you're quick! I'll take a look.

Comment (garrettwrong) [reply src/aspire/utils/rotation.py]: Updated.

One of the test I avoid the gimble lock (most of the time I guess) by using a random rotation.

For the others there is an option to disable now.

File: src/aspire/utils/rotation.py [hunk 2]
```diff
@@ -13,11 +13,13 @@
...
 
 class Rotation:
-    def __init__(self, matrices):
+    def __init__(self, matrices, gimble_lock_warnings=True):
         """
          Initialize a Rotation object
 
         :param matrices: Rotation matrices to initialize Rotation object.
+        :param gimble_lock_warnings: Optionally show gimble lock warnings.
+             Defaults to `True`, setting to `False` filters warnings.
```
Comment (j-c-c) [inline src/aspire/utils/rotation.py:22]: Nice!

File: src/aspire/utils/rotation.py [hunk 3]
```diff
@@ -31,6 +33,7 @@ def __init__(self, matrices):
         # Attribute indicating if rotation matrices are about the z-axis,
         # which will cause a gimbal lock warning for ZYZ convention.
         self._about_z = False
```
Comment (j-c-c) [inline src/aspire/utils/rotation.py]: I think we can drop this flag and its use on line 299.

**Review by j-c-c — CHANGES_REQUESTED** [review]
Looks good! Just one small change. NGDGU lol

File: src/aspire/utils/rotation.py (hunk 2 above)
Comment (garrettwrong) [reply src/aspire/utils/rotation.py:22]: image

File: src/aspire/utils/rotation.py (hunk 3 above)
Comment (garrettwrong) [reply src/aspire/utils/rotation.py]: sure, just pushed.

**Review by j-c-c — APPROVED** [review]
🚀

File: src/aspire/utils/rotation.py (hunk 2 above)
Comment (j-c-c) [reply src/aspire/utils/rotation.py:22]: 🤣🤣🤣

**Comment by garrettwrong** [comment]
Merging.

**Pull request merged**